packages/@aws-cdk/aws-autoscaling/lib/auto-scaling-group.ts

```diff
@@ -584,6 +678,10 @@ export class AutoScalingGroup extends AutoScalingGroupBase implements
       loadBalancerNames: Lazy.listValue({ produce: () => this.loadBalancerNames }, { omitEmpty: true }),
       targetGroupArns: Lazy.listValue({ produce: () => this.targetGroupArns }, { omitEmpty: true }),
       notificationConfigurations: this.renderNotificationConfiguration(),
+      metricsCollection: Lazy.anyValue({ produce: () => this.metricsCollections.length === 0 ? undefined : this.metricsCollections.map(mc => ({
+        granularity: '1Minute',
+        metrics: mc.metrics?.length !== 0 ? mc.metrics.map(m => m.name) : undefined,
+      }))}),
```

Comment:
This is hard to read. Let's separate this into a private method and use intermediate variables to make this more readable.